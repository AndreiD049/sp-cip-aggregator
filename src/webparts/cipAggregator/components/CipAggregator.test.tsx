import { mount, shallow } from 'enzyme';
import React from 'react';
import CipAggregator from './CipAggregator';

test('Hello', () => {
    const component = shallow(
        <CipAggregator 
            description='Hello world'
            environmentMessage='environment'
            isDarkTheme
            userDisplayName='Andrei'
            hasTeamsContext={false}
        />
    );

    const header = <h2>Well done, Andrei!</h2>;
    expect(component.html()).toMatchSnapshot()
    expect(component.contains(header)).toBeTruthy();
    expect(component.find("[data-testid='description']").text())
        .toBe("Web part property value: Hello world")
});