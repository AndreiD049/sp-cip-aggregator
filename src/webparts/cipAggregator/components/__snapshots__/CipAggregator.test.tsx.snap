// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hello 1`] = `"<section class=\\"undefined \\"><div><img alt=\\"\\" src=\\"test-file-stub\\"/><h2>Well done, Andrei!</h2><div>environment</div><div data-testid=\\"description\\">Web part property value: <strong>Hello world</strong></div></div><div><h3>Welcome to SharePoint Framework!</h3><p>The SharePoint Framework (SPFx) is a extensibility model for Microsoft Viva, Microsoft Teams and SharePoint. It&#x27;s the easiest way to extend Microsoft 365 with automatic Single Sign On, automatic hosting and industry standard tooling.</p><h4>Learn more about SPFx development:</h4><ul><li><a href=\\"https://aka.ms/spfx\\" target=\\"_blank\\">SharePoint Framework Overview</a></li><li><a href=\\"https://aka.ms/spfx-yeoman-graph\\" target=\\"_blank\\">Use Microsoft Graph in your solution</a></li><li><a href=\\"https://aka.ms/spfx-yeoman-teams\\" target=\\"_blank\\">Build for Microsoft Teams using SharePoint Framework</a></li><li><a href=\\"https://aka.ms/spfx-yeoman-viva\\" target=\\"_blank\\">Build for Microsoft Viva Connections using SharePoint Framework</a></li><li><a href=\\"https://aka.ms/spfx-yeoman-store\\" target=\\"_blank\\">Publish SharePoint Framework applications to the marketplace</a></li><li><a href=\\"https://aka.ms/spfx-yeoman-api\\" target=\\"_blank\\">SharePoint Framework API reference</a></li><li><a href=\\"https://aka.ms/m365pnp\\" target=\\"_blank\\">Microsoft 365 Developer Community</a></li></ul></div></section>"`;
